modules/bcftools/annotate/main.nf

```diff
@@ -0,0 +1,42 @@
+process BCFTOOLS_ANNOTATE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.15" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.15--haf5b3da_0':
+        'quay.io/biocontainers/bcftools:1.15--haf5b3da_0' }"
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), path("*_annotated.vcf.gz"), optional:true , emit: vcf
+    tuple val(meta), path("*_annotated.bcf")   , optional:true , emit: bcf
+    path "versions.yml"                                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def matcher = input =~ /vcf/
+    def output_suffix = matcher ? "vcf.gz" : "bcf"
```

Comment:
```suggestion
    def output_suffix = input ==~ /\S+\.*vcf\.\S*/ ? "vcf.gz" : "bcf"
```
I don't see `matcher` used anywhere else so just use it in the ternary operator.
ps. difference between match and find operator: https://groovy-lang.org/operators.html#_match_operator.
Also a match is from the beginning to the end of a string, so provide a full pattern.